app/main.py

```diff
@@ -0,0 +1,137 @@
+from typing import Type
+from copy import deepcopy
+
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel, Field, UUID4
+
+from models import Secret
+
+
+app = FastAPI()
+
+
+@app.get("/")
+def read_root():
+    return {"Hello": "World"}
+
+
+IN_MEMORY_DB = dict()
+
+
+class NotFoundException(Exception):
+    def __init__(self):
+        super().__init__("Secret with provided ID does not exist")
+
+
+class NotCreatedException(Exception):
+    def __init__(self):
+        super().__init__("Error creating secret")
+
+
+class SecretRepository:
+    def getSecretOrEexception(self, id):
+        secret = IN_MEMORY_DB.get(id)
+        if not secret:
+            raise NotFoundException()
+
+        return deepcopy(secret)
+
+    def insertSecret(self, secret: str) -> Type[Secret]:
+        try:
+            newSecret = Secret(secret)
+            IN_MEMORY_DB[newSecret.id] = newSecret
+
+            return self.getSecretOrEexception(newSecret.id)
+        except NotFoundException:
+            raise NotCreatedException()
+
+    def getSecret(self, id: UUID4, meta: bool = False):
+        try:
+            secret = self.getSecretOrEexception(id)
+            if meta:
+                return secret
+            return secret.removeMeta()
+
+        except NotFoundException as err:
+            raise err
+
+    def getSecretInfo(self, id: UUID4):
+        pass
+
+    def updateSecret(self, id: UUID4, secret: str):
+        try:
+            oldSecret = self.getSecretOrEexception(id)
+            IN_MEMORY_DB[id] = oldSecret.updateSecret(secret)
+            return self.getSecretOrEexception(id)
+        except NotFoundException as err:
+            raise err
+
+    def invalidateSecret(self, id: UUID4):
+        try:
+            secret = self.getSecretOrEexception(id)
+            secret.deactivate()
+            IN_MEMORY_DB[secret.id] = secret
+            return "OK"
+        except NotFoundException as err:
+            raise err
+
+
```

Comment:
can we use a factory method/function inside repository.py (create_repository) this will avoid the duplication, Are you sure that we need an unique repository? Can we try to use a Python context manager to initialize and close the repository as soon we use?